### ISSUE TITLE
Incorrect failure for UselessSetterCall when local variable is a reference

### ISSUE THREAD

**Issue by jeremywadsack** [rubocop-hq/rubocop#5420]
In the following method, Rubocop incorrectly fails the `UselessSetterCall` cop:

```
      def reference_assignment(list)
        list ||= []
        entry = list.first
        list << (entry = {}) unless entry
        entry["value"] = true
      end
```

--------

## Expected behavior

RuboCop should be able to track the fact that `entry` is a local _reference_ to a value and that the reference is added to a non-local scope.

## Actual behavior

The error message flags the local variable `entry` with the following.

```
W: Lint/UselessSetterCall: Useless setter call to local variable entry.
        entry["value"] = true
        ^^^^^
```

Note that if I alter the above code to look like this, then it doesn't get flagged:

```
      def reference_assignment(list)
        list ||= []
        entry = list.first
        list << {} unless entry
        entry = list.first
        entry["value"] = true
      end
```

While I tried to simplify this example as much as possible, in my actual code `list.first` is more complicated and therefore I'd rather not repeat the look-up just to make Rubocop happy.

## Steps to reproduce the problem

See above.

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by jeremywadsack** [comment]
In rubocop 0.68.1 this is still failing.

**Comment by jeremywadsack** [comment]
Closing this as it appears fixed with Rubocop 0.74
